### ISSUE TITLE
Unable to change the default remove behaviour

### ISSUE THREAD

**Issue by aitorllj93** [pqina/filepond#47]
Hi,

I have troubles handing de DELETE file method

The revert handler works well when upload is done with the component, but when I load 
the files preloaded with this format:

```js
{
  source: file,
  options: {
    type: 'local'
  }
}
```

I'm unable to change the remove file default behaviour

Thanks for your time

**Comment by rikschennink** [comment]
I’ll test it this afternoon, it’s set up to remove temporary files, so maybe I missed a use-case 🙃

**Comment by rikschennink** [comment]
@aitorllj93 There's currently no `remove` end-point. It only handles reverting temporary files.

The reasoning behind this is that destructive actions like that should be done after careful consideration ( and would require an undo option or serious confirm popup ). FilePond keeps a list of current files, then, when the parent form is submitted you can determine the differences after the user made changes to the files list and remove the necessary files from the server.

I'm considering adding this as an option but am wondering if the solution above would work for your use-case?

**Comment by aitorllj93** [comment]
Hi @rikschennink, sorry, was my fault, I had already handled the remove event but didn't saw it.

The problem now is, this event gets called also during revert process so I'm unable to know if that file was already deleted by the revert handler

**Comment by rikschennink** [comment]
@aitorllj93 Yes that's also a way handle it. I think a property to indicate the origins of the file would be a nice addition, something like: `local`, `limbo`, `input`. That should offer enough detail to be able to handle file removal in the remove handler.

**Comment by rikschennink** [comment]
@aitorllj93 Just added the `origin` property to the file item and the `FileOrigin` enum to the FilePond object. The origin is either 1, 2 or 3, and you can use the FileOrigin enum to make your code more readable.

It's available in version `1.6.1`

```js
export const FileOrigin = {
    INPUT:1,
    LIMBO:2,
    LOCAL:3
};
```
